arrow-site/docs/docs/fx/purity-and-referentially-transparent-functions/README.MD

```diff
@@ -0,0 +1,131 @@
+
+```kotlin
+suspend fun sayGoodBye(): Unit =
+  println("Good bye World!")
+
+suspend fun sayHello(): Unit =
+  println("Hello World")
+
+suspend fun greet(): Unit {
+  sayHello() // this is ok because
+  sayGoodBye() // `greet` is also `suspend`
+}
+```
+
+## Executing effectful programs
+
+The `greet` program is ready to run as soon as the user is ready to commit to an execution strategy that is either `blocking` or `non-blocking`.
+`blocking` execution strategies will block the current thread that's waiting for the program to yield a value, whereas `non-blocking` strategies will immediately return and perform the program's work without blocking the current thread.
+Since both blocking and non-blocking execution scenarios perform side effects, we consider running effects as an `unsafe` operation.
+
+KotlinX Coroutines offers `runBlocking`, `launch` and `async` to run a suspended program.
+Usage of unsafe runner functions (like `runBlocking` in this case) is reserved for the end of the world and may be the only impure execution of a well-typed functional program.
+`launch` and `async` can however also be used in pure/safe ways if they're used within another `suspend fun`.
+
+```kotlin:ank:playground
+import kotlinx.coroutines.*
+import arrow.fx.coroutines.*
+
+//sampleStart
+suspend fun sayHello(): Unit =
+  println("Hello World")
+
+suspend fun sayGoodBye(): Unit =
+  println("Good bye World!")
+
+suspend fun greet(): Unit {
+  sayHello()
+  sayGoodBye()
+}
+
+fun main(): Unit = runBlocking { // The edge of our world
+  greet()
+}
+//sampleEnd
+```
+
+Arrow Fx emphasizes the guarantee that users understand when they are performing side effects in their program declaration.
+
+If you're not very familiar with Functional Programming, and you've made it this far, you may realize that, despite the buzzwords and some FP jargon, you already know how to use Arrow Fx in general.
+This is because Arrow Fx brings the most popular imperative style to effectful programs with few simple primitives for effect control and asynchronous programming.
+
+# Conclusion
+
+Arrow Fx offers an idiomatic way of doing effectfull FP with Kotlin's coroutine system.
+It does this by providing support for cancellation, error handling, resource handling and all goodies you might be familiar with from other functional effect systems.
+
+Arrow Fx offers direct style syntax and effect control without compromises, and removes the syntactic burden of type parametrization while still yielding programs that are pure, safe, and referentially transparent.
+
+Despite some limitations of the Kotlin type system, like lack of Higher Kinded Types, the Kotlin language is excellent for encoding typed programs with a first-class imperative syntax for FP.
+These programs are approachable by the broader programming community, and arguably easier to encode than some solutions currently used in mainstream Kotlin and Scala FP communities.
```

Comment:
```suggestion
```
I don't see a need to compare in the docs like this anymore. I think we can get rid of this.